### PR TITLE
PL-1309 adding setCustomerInitiated method

### PR DESCRIPTION
**Pull request by VKCR**

### CODE REVIEW

File: src/sample/RedeemAccessTokenSample.js [hunk 1]
```diff
@@ -19,6 +19,7 @@ class RedeemAccessTokenSample {
         const accounts = await grantee.getAccounts();
 
         // Get informtion we want:
+        grantee.setCustomerInitiated();
```
Comment (zixunjian) [inline src/sample/RedeemAccessTokenSample.js]: should this be get?

File: src/http/AuthHeader.js [hunk 2]
```diff
@@ -96,6 +97,17 @@ class AuthHeader {
         }
         return config.KeyLevel.LOW;
     }
+
+    static _customerInitiated(context) {
+        // if the customer initiated request flag is set to true, we add it to the header, and reset the flag.
+        if (context !== undefined &&
+                context.customerInitiated !== undefined &&
+                context.customerInitiated === true) {
```
Comment (zixunjian) [inline src/http/AuthHeader.js]: seems redundant
`if (context && context.customerInitiated)` ?

Comment (VKCR) [reply src/http/AuthHeader.js]: I'm not sure how js works. It won't scream at me if context.customerInitiated is undefined?

File: src/sample/RedeemAccessTokenSample.js (hunk 1 above)
Comment (VKCR) [reply src/sample/RedeemAccessTokenSample.js]: no, only set

File: src/http/AuthHeader.js (hunk 2 above)
Comment (zixunjian) [reply src/http/AuthHeader.js]: if context.customerInitiated is undefined, the `if` won't go through. `undefined` evaluates to `false`

Comment (VKCR) [reply src/http/AuthHeader.js]: ok, changed.

File: src/http/AuthContext.js [hunk 3]
```diff
@@ -10,6 +10,7 @@ class AuthContext {
     constructor() {
         this._onBehalfOf = undefined;
         this._keyLevel = config.KeyLevel.LOW;
+        this._customerInitiated = undefined;
```
Comment (mariano54) [inline src/http/AuthContext.js]: why not set it to true? It's unclear what undefined here means

Comment (VKCR) [reply src/http/AuthContext.js]: should not be true by default

Comment (zixunjian) [reply src/http/AuthContext.js]: what about setting it to false then? I agree undefined seems strange

**Pull request merged**